python/console/console_base.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding:utf-8 -*-
+"""
+/***************************************************************************
+Python Console for QGIS
+                             -------------------
+begin                : 2020-06-04
+copyright            : (C) 2020 by Richard Duivenvoorde
+email                : Richard Duivenvoorde (at) duif (dot) net
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+Some portions of code were taken from https://code.google.com/p/pydee/
+"""
+
+from qgis.PyQt.QtCore import Qt, QUrl
+from qgis.PyQt.QtGui import QColor, QFontDatabase, QDesktopServices
+from qgis.PyQt.Qsci import QsciScintilla, QsciLexerPython, QsciAPIs
+from qgis.core import QgsApplication, Qgis
+import os
+
+
+class QgsQsciScintillaBase(QsciScintilla):
```

Comment:
Could we come up with a better class name here? QgsQsci isn't super great. But I have no alternative to offer so if no better suggestion emerges, that can stick :)

Reply:
Happy to rename, just wanted to follow the normal naming conventions...

Reply:
Fine.